feat(dolt_branch): Add Dolt branch module

Perniciosius opened the pull request:
#### Description
Add new module `dolt_branch`, which shows active branch of dolt repo in current directory.
#### Motivation and Context
Refer #3353 

#### Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/45752299/148426060-5cefd48e-c575-435c-8702-c01757176404.png)


#### How Has This Been Tested?
- [ ] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:
- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

Perniciosius committed:
add module config

Perniciosius committed:
add module

Perniciosius committed:
update required files

Perniciosius committed:
update docs

moko256 committed:
feat(elvish): last command status (#3403)

Commit:
docs(faq): for uninstalling, change which to command -v (#3394)

* Changed 'which' to 'command -v' which is more general, and available on more systems

* docs: for uninstalling, change which to command -v

rashil2000 committed:
feat: add support for cmd (#3277)

* feat: add support for cmd

* add preprompt and precmd support

* add keymap support

* add info about minimum Clink version

* simplify escaping

* add handling for cmd custom commands

* add support for transient_prompt and transient_rprompt

* Revert 914057952508e81e20086fcb707ba2a0be85fdd3

This reverts commit "add support for transient_prompt and transient_rprompt"

* Apply suggestions from code review

* disable cmd shell custom commands

* any shell other than cmd can be used

* better error and correct script location

* move shell check in `map_no_escaping`

dependabot[bot] committed:
build: bump toml_edit from 0.12.3 to 0.12.4 (#3412)

Bumps [toml_edit](https://github.com/ordian/toml_edit) from 0.12.3 to 0.12.4.
- [Release notes](https://github.com/ordian/toml_edit/releases)
- [Changelog](https://github.com/ordian/toml_edit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ordian/toml_edit/compare/v0.12.3...v0.12.4)

---
updated-dependencies:
- dependency-name: toml_edit
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump indexmap from 1.7.0 to 1.8.0 (#3411)

Bumps [indexmap](https://github.com/bluss/indexmap) from 1.7.0 to 1.8.0.
- [Release notes](https://github.com/bluss/indexmap/releases)
- [Changelog](https://github.com/bluss/indexmap/blob/master/RELEASES.rst)
- [Commits](https://github.com/bluss/indexmap/compare/1.7.0...1.8.0)

---
updated-dependencies:
- dependency-name: indexmap
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump versions from 3.0.3 to 4.0.0 (#3410)

Bumps [versions](https://github.com/fosskers/rs-versions) from 3.0.3 to 4.0.0.
- [Release notes](https://github.com/fosskers/rs-versions/releases)
- [Changelog](https://github.com/fosskers/rs-versions/blob/master/CHANGELOG.md)
- [Commits](https://github.com/fosskers/rs-versions/commits/v4.0.0)

---
updated-dependencies:
- dependency-name: versions
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump clap from 3.0.5 to 3.0.6 (#3415)

Bumps [clap](https://github.com/clap-rs/clap) from 3.0.5 to 3.0.6.
- [Release notes](https://github.com/clap-rs/clap/releases)
- [Changelog](https://github.com/clap-rs/clap/blob/master/CHANGELOG.md)
- [Commits](https://github.com/clap-rs/clap/compare/v3.0.5...v3.0.6)

---
updated-dependencies:
- dependency-name: clap
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

moko256 committed:
docs(cmd_duration): rust-notify -> notify-rust (#3417)

dependabot[bot] committed:
build: bump clap_complete from 3.0.2 to 3.0.3 (#3422)

Bumps [clap_complete](https://github.com/clap-rs/clap) from 3.0.2 to 3.0.3.
- [Release notes](https://github.com/clap-rs/clap/releases)
- [Changelog](https://github.com/clap-rs/clap/blob/master/CHANGELOG.md)
- [Commits](https://github.com/clap-rs/clap/compare/clap_complete-v3.0.2...clap_complete-v3.0.3)

---
updated-dependencies:
- dependency-name: clap_complete
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump toml_edit from 0.12.4 to 0.12.6 (#3420)

Bumps [toml_edit](https://github.com/ordian/toml_edit) from 0.12.4 to 0.12.6.
- [Release notes](https://github.com/ordian/toml_edit/releases)
- [Changelog](https://github.com/ordian/toml_edit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ordian/toml_edit/compare/v0.12.4...v0.12.6)

---
updated-dependencies:
- dependency-name: toml_edit
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump clap from 3.0.6 to 3.0.7 (#3421)

Bumps [clap](https://github.com/clap-rs/clap) from 3.0.6 to 3.0.7.
- [Release notes](https://github.com/clap-rs/clap/releases)
- [Changelog](https://github.com/clap-rs/clap/blob/master/CHANGELOG.md)
- [Commits](https://github.com/clap-rs/clap/compare/v3.0.6...v3.0.7)

---
updated-dependencies:
- dependency-name: clap
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump toml_edit from 0.12.6 to 0.13.0 (#3431)

Bumps [toml_edit](https://github.com/ordian/toml_edit) from 0.12.6 to 0.13.0.
- [Release notes](https://github.com/ordian/toml_edit/releases)
- [Changelog](https://github.com/ordian/toml_edit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ordian/toml_edit/compare/v0.12.6...v0.13.0)

---
updated-dependencies:
- dependency-name: toml_edit
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

davidkna committed:
fix(bug-report): remove git.io link shortening (#3425)

matchai committed:
chore(release): v1.2.0

segevfiner committed:
fix(pwsh): Avoid polluting the global function namespace (#3424)

* fix(pwsh): Avoid polluting the global function namespace

This is an alternative to #3386

* Review changes

* Move continuation prompt after setting starship env

davidkna committed:
fix: allow compilation without battery feature (#3435)

davidkna committed:
fix(properties): fix regressions in `status`, `pipestatus` and `terminal-width` handling (#3399)

BRAVO68WEB committed:
docs: Fixed starship config location (#3432)

* Fixed starship config location

config file during creation and when used as export variable into shells are different.
Fixed to make it uniform and avoid errors

* Update README.md

* Update docs/config/README.md

Co-authored-by: Thomas O'Donnell <andytom@users.noreply.github.com>

* Update docs/config/README.md

Co-authored-by: Thomas O'Donnell <andytom@users.noreply.github.com>

* Update docs/config/README.md

Co-authored-by: Thomas O'Donnell <andytom@users.noreply.github.com>

Co-authored-by: Thomas O'Donnell <andytom@users.noreply.github.com>

matchai committed:
chore(release): v1.2.1

matchai committed:
docs(i18n): new Crowdin updates (#3344)

* New translations README.md (Portuguese)

* New translations README.md (Spanish)

* New translations README.md (Polish)

* New translations README.md (Dutch)

* New translations README.md (Korean)

* New translations README.md (Japanese)

* New translations README.md (Italian)

* New translations README.md (German)

* New translations README.md (Arabic)

* New translations README.md (Sorani (Kurdish))

* New translations README.md (French)

* New translations README.md (Russian)

* New translations README.md (Indonesian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Vietnamese)

* New translations README.md (Chinese Traditional)

* New translations README.md (Chinese Simplified)

* New translations README.md (Turkish)

* New translations README.md (Portuguese)

* New translations README.md (Spanish)

* New translations README.md (Polish)

* New translations README.md (Dutch)

* New translations README.md (Korean)

* New translations README.md (Japanese)

* New translations README.md (Italian)

* New translations README.md (German)

* New translations README.md (Arabic)

* New translations README.md (Sorani (Kurdish))

* New translations README.md (French)

* New translations README.md (French)

* New translations README.md (French)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (French)

* New translations README.md (Russian)

* New translations README.md (Indonesian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Vietnamese)

* New translations README.md (Chinese Traditional)

* New translations README.md (Chinese Simplified)

* New translations README.md (Turkish)

* New translations README.md (Portuguese)

* New translations README.md (Spanish)

* New translations README.md (Polish)

* New translations README.md (Dutch)

* New translations README.md (Korean)

* New translations README.md (Japanese)

* New translations README.md (Italian)

* New translations README.md (German)

* New translations README.md (Arabic)

* New translations README.md (Sorani (Kurdish))

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (French)

* New translations README.md (French)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (French)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (French)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (French)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (French)

* New translations README.md (Russian)

* New translations README.md (Indonesian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Vietnamese)

* New translations README.md (Chinese Traditional)

* New translations README.md (Chinese Simplified)

* New translations README.md (Turkish)

* New translations README.md (Portuguese)

* New translations README.md (Spanish)

* New translations README.md (Polish)

* New translations README.md (Dutch)

* New translations README.md (Korean)

* New translations README.md (Japanese)

* New translations README.md (Italian)

* New translations README.md (German)

* New translations README.md (Arabic)

* New translations README.md (Sorani (Kurdish))

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (French)

* New translations README.md (Russian)

* New translations README.md (Indonesian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Vietnamese)

* New translations README.md (Chinese Traditional)

* New translations README.md (Chinese Simplified)

* New translations README.md (Turkish)

* New translations README.md (Portuguese)

* New translations README.md (Spanish)

* New translations README.md (Polish)

* New translations README.md (Dutch)

* New translations README.md (Korean)

* New translations README.md (Japanese)

* New translations README.md (Italian)

* New translations README.md (German)

* New translations README.md (Arabic)

* New translations README.md (Sorani (Kurdish))

* New translations README.md (French)

* New translations README.md (Korean)

* New translations README.md (German)

* New translations README.md (Italian)

* New translations README.md (Italian)

* New translations README.md (Japanese)

* New translations README.md (Japanese)

* New translations README.md (Korean)

* New translations README.md (Dutch)

* New translations README.md (Dutch)

* New translations README.md (Polish)

* New translations README.md (Polish)

* New translations README.md (Portuguese)

* New translations README.md (Portuguese)

* New translations README.md (Russian)

* New translations README.md (German)

* New translations README.md (Arabic)

* New translations README.md (Spanish)

* New translations README.md (Chinese Simplified)

* New translations README.md (Arabic)

* New translations README.md (German)

* New translations README.md (Italian)

* New translations README.md (Japanese)

* New translations README.md (Korean)

* New translations README.md (Dutch)

* New translations README.md (Polish)

* New translations README.md (Portuguese)

* New translations README.md (Russian)

* New translations README.md (Turkish)

* New translations README.md (Chinese Traditional)

* New translations README.md (Arabic)

* New translations README.md (Vietnamese)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Indonesian)

* New translations README.md (Sorani (Kurdish))

* New translations README.md (French)

* New translations README.md (French)

* New translations README.md (Spanish)

* New translations README.md (Spanish)

* New translations README.md (Russian)

* New translations README.md (French)

* New translations README.md (Dutch)

* New translations README.md (Spanish)

* New translations README.md (Arabic)

* New translations README.md (German)

* New translations README.md (Italian)

* New translations README.md (Japanese)

* New translations README.md (Korean)

* New translations README.md (Polish)

* New translations README.md (Portuguese)

* New translations README.md (Russian)

* New translations README.md (Turkish)

* New translations README.md (Chinese Traditional)

* New translations README.md (Vietnamese)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Indonesian)

* New translations README.md (Chinese Simplified)

* New translations README.md (Sorani (Kurdish))

* New translations README.md (Turkish)

* New translations README.md (Turkish)

* New translations README.md (Chinese Simplified)

* New translations README.md (Chinese Simplified)

* New translations README.md (Chinese Traditional)

* New translations README.md (Chinese Traditional)

* New translations README.md (Vietnamese)

* New translations README.md (Sorani (Kurdish))

* New translations README.md (Vietnamese)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Indonesian)

* New translations README.md (Indonesian)

* New translations README.md (Sorani (Kurdish))

* New translations README.md (Spanish)

* New translations README.md (Spanish)

* New translations README.md (Spanish)

* New translations README.md (Russian)

* New translations README.md (Russian)

* New translations README.md (French)

* New translations README.md (Russian)

* New translations README.md (Indonesian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Vietnamese)

* New translations README.md (Chinese Traditional)

* New translations README.md (Chinese Simplified)

* New translations README.md (Turkish)

* New translations README.md (Portuguese)

* New translations README.md (Spanish)

* New translations README.md (Polish)

* New translations README.md (Dutch)

* New translations README.md (Korean)

* New translations README.md (Japanese)

* New translations README.md (Italian)

* New translations README.md (German)

* New translations README.md (Arabic)

* New translations README.md (Sorani (Kurdish))

* New translations README.md (Chinese Simplified)

* New translations README.md (Chinese Simplified)

* New translations README.md (Chinese Simplified)

* New translations README.md (Chinese Simplified)

* New translations README.md (Chinese Simplified)

* New translations README.md (Chinese Simplified)

* New translations README.md (Chinese Simplified)

* New translations README.md (Chinese Simplified)

* New translations README.md (Chinese Simplified)

* New translations README.md (French)

* New translations README.md (Russian)

* New translations README.md (Indonesian)

* New translations README.md (Portuguese, Brazilian)

* New translations README.md (Vietnamese)

* New translations README.md (Chinese Traditional)

* New translations README.md (Chinese Simplified)

* New translations README.md (Turkish)

* New translations README.md (Portuguese)

* New translations README.md (Spanish)

* New translations README.md (Polish)

* New translations README.md (Dutch)

* New translations README.md (Korean)

* New translations README.md (Japanese)

* New translations README.md (Italian)

* New translations README.md (German)

* New translations README.md (Arabic)

* New translations README.md (Sorani (Kurdish))

segevfiner committed:
feat(pwsh): Set STARSHIP_SHELL to pwsh on PS >5 (#3443)

* feat(pwsh): Set STARSHIP_SHELL to pwsh ON PS >5

This is needed for the custom module to use the correct shell.

Closes #3376

* Update src/context.rs

Co-authored-by: David Knaack <davidkna@users.noreply.github.com>

Co-authored-by: David Knaack <davidkna@users.noreply.github.com>

mandx committed:
ci: Add jobs for `cargo check` with all features and no features (#3440)

* ci: Add jobs for `cargo check` with all features and no features

Augmenting #3435: These jobs will check that compilation still succeeds when no features are selected and with all features selected. So when in the future new features are added, these quick checks can prevent miss-compilation for users that like to tweak with feature sets.

* (fixup) More descriptive job names

* ci: Make the new `cargo_check_*` jobs depend on `cargo_check`

dependabot[bot] committed:
build: bump gethostname from 0.2.1 to 0.2.2 (#3453)

Bumps gethostname from 0.2.1 to 0.2.2.

---
updated-dependencies:
- dependency-name: gethostname
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump os_info from 3.0.9 to 3.1.0 (#3452)

Bumps [os_info](https://github.com/stanislav-tkach/os_info) from 3.0.9 to 3.1.0.
- [Release notes](https://github.com/stanislav-tkach/os_info/releases)
- [Changelog](https://github.com/stanislav-tkach/os_info/blob/master/CHANGELOG.md)
- [Commits](https://github.com/stanislav-tkach/os_info/commits)

---
updated-dependencies:
- dependency-name: os_info
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump serde_json from 1.0.74 to 1.0.75 (#3451)

Bumps [serde_json](https://github.com/serde-rs/json) from 1.0.74 to 1.0.75.
- [Release notes](https://github.com/serde-rs/json/releases)
- [Commits](https://github.com/serde-rs/json/compare/v1.0.74...v1.0.75)

---
updated-dependencies:
- dependency-name: serde_json
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump process_control from 3.2.0 to 3.2.1 (#3450)

Bumps [process_control](https://github.com/dylni/process_control) from 3.2.0 to 3.2.1.
- [Release notes](https://github.com/dylni/process_control/releases)
- [Commits](https://github.com/dylni/process_control/compare/3.2.0...3.2.1)

---
updated-dependencies:
- dependency-name: process_control
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump clap_complete from 3.0.3 to 3.0.4 (#3449)

Bumps [clap_complete](https://github.com/clap-rs/clap) from 3.0.3 to 3.0.4.
- [Release notes](https://github.com/clap-rs/clap/releases)
- [Changelog](https://github.com/clap-rs/clap/blob/master/CHANGELOG.md)
- [Commits](https://github.com/clap-rs/clap/compare/clap_complete-v3.0.3...clap_complete-v3.0.4)

---
updated-dependencies:
- dependency-name: clap_complete
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump clap from 3.0.7 to 3.0.9 (#3456)

Bumps [clap](https://github.com/clap-rs/clap) from 3.0.7 to 3.0.9.
- [Release notes](https://github.com/clap-rs/clap/releases)
- [Changelog](https://github.com/clap-rs/clap/blob/master/CHANGELOG.md)
- [Commits](https://github.com/clap-rs/clap/compare/v3.0.7...v3.0.9)

---
updated-dependencies:
- dependency-name: clap
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

matchai committed:
docs: break installation instructions to collapsable sections  (#3448)

As the number of supported systems and shells has grown, we've had to hand-pick what installation methods would be the "blessed" ones to show on our README. This change changes that to include all installation methods, but hides them each into OS and shell-specific collapsed sections, as is done by the excellent ajeetdsouza/zoxide.

dependabot[bot] committed:
build: bump clap from 3.0.9 to 3.0.10 (#3463)

Bumps [clap](https://github.com/clap-rs/clap) from 3.0.9 to 3.0.10.
- [Release notes](https://github.com/clap-rs/clap/releases)
- [Changelog](https://github.com/clap-rs/clap/blob/master/CHANGELOG.md)
- [Commits](https://github.com/clap-rs/clap/compare/v3.0.9...v3.0.10)

---
updated-dependencies:
- dependency-name: clap
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

chipbuster committed:
ci: Use `dprint` to format documentation + TOML files (#3426)

Adds Rust-based standardized markdown formatting and a CI step to catch unformatted files.

moko256 committed:
perf(elvish): Use built-in `randint` instead of `starship session`. (#3479)

haraldh committed:
feat: add a container indicator (#3304)

* test: add mock method for absolute files

Signed-off-by: Harald Hoyer <harald@hoyer.xyz>

* feat(module): add a container indicator module

Adds a container type indicator, if inside a container,
detected via the presence of some marker files.

E.g. inside a podman container entered with `toolbox enter`
the prompt changes to the container name and version.

```
starship on  container_rebased [$!] is 📦 v1.0.0 via 🦀 v1.56.1
❯ toolbox enter

starship on  container_rebased [$!] is 📦 v1.0.0 via 🦀 v1.56.1
⬢ [fedora-toolbox:35] ❯
```

Signed-off-by: Harald Hoyer <harald@hoyer.xyz>

chipbuster committed:
feat(install): Add posix shell check (#3474)

* feat(install): Add posix shell check

* Change function name

chipbuster committed:
ci: Block changes to crowdin-generated files (#3427)

* ci: Add workflow to block crowdin changes

* Add a change which should not block the PR (will be reverted later)

* Repair broken workflows

* Add a change which should stop the PR

* Revert "Add a change which should not block the PR (will be reverted later)"

This reverts commit eafb5eaed1c2ec4909282990d51b7eb4d7a91fdc.

* Revert "Add a change which should stop the PR"

This reverts commit bfe6e7a39d5559dd6f58fd8cff854ed53948fb40.

jsnider-mtu committed:
docs: Add set commands to advanced config doc (#1526)

andytom committed:
ci: Fix failing CI jobs (#3481)

Have updated the workflow to only run the "Block Translated Changes" job
on PRs since it will fail when run on pushes to a branch. Have fixed the
formatting while I was here.

chipbuster committed:
ci: Fix aws::expiration_date_set_from_file race (#3484)

* ci: Fix aws::expiration_date_set_from_file race

While aws::expiration_date_set_from_file will almost-always work
perfectly locally, it is theoretically possible for the thread running
the test to be scheduled away betwen writing the file with timing
information and then actually reading it, resulting in a
shorter-than-expected time appearing in the module. This can also happen
if the test triggers right at the very end of a second (e.g. at
10:45:47.999995).

This appears to actually happen sometimes on heavily-loaded GitHub
Actions runners. To fix this issue, we allow for up to a two-second
delay between when the file is written and when the test actually fires
(allowing "30m", "29m59s", and "29m58s" as possible values).

* Fix typo

dependabot[bot] committed:
build: bump serde from 1.0.133 to 1.0.135 (#3496)

Bumps [serde](https://github.com/serde-rs/serde) from 1.0.133 to 1.0.135.
- [Release notes](https://github.com/serde-rs/serde/releases)
- [Commits](https://github.com/serde-rs/serde/compare/v1.0.133...v1.0.135)

---
updated-dependencies:
- dependency-name: serde
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump @vuepress/plugin-google-analytics in /docs (#3495)

Bumps [@vuepress/plugin-google-analytics](https://github.com/vuejs/vuepress/tree/HEAD/packages/@vuepress/plugin-google-analytics) from 1.9.5 to 1.9.7.
- [Release notes](https://github.com/vuejs/vuepress/releases)
- [Changelog](https://github.com/vuejs/vuepress/blob/master/CHANGELOG.md)
- [Commits](https://github.com/vuejs/vuepress/commits/v1.9.7/packages/@vuepress/plugin-google-analytics)

---
updated-dependencies:
- dependency-name: "@vuepress/plugin-google-analytics"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump clap from 3.0.10 to 3.0.12 (#3506)

Bumps [clap](https://github.com/clap-rs/clap) from 3.0.10 to 3.0.12.
- [Release notes](https://github.com/clap-rs/clap/releases)
- [Changelog](https://github.com/clap-rs/clap/blob/master/CHANGELOG.md)
- [Commits](https://github.com/clap-rs/clap/compare/v3.0.10...v3.0.12)

---
updated-dependencies:
- dependency-name: clap
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump serde_json from 1.0.75 to 1.0.78 (#3509)

Bumps [serde_json](https://github.com/serde-rs/json) from 1.0.75 to 1.0.78.
- [Release notes](https://github.com/serde-rs/json/releases)
- [Commits](https://github.com/serde-rs/json/compare/v1.0.75...v1.0.78)

---
updated-dependencies:
- dependency-name: serde_json
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump which from 4.2.2 to 4.2.4 (#3507)

Bumps [which](https://github.com/harryfei/which-rs) from 4.2.2 to 4.2.4.
- [Release notes](https://github.com/harryfei/which-rs/releases)
- [Commits](https://github.com/harryfei/which-rs/compare/4.2.2...4.2.4)

---
updated-dependencies:
- dependency-name: which
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump clap_complete from 3.0.4 to 3.0.5 (#3508)

Bumps [clap_complete](https://github.com/clap-rs/clap) from 3.0.4 to 3.0.5.
- [Release notes](https://github.com/clap-rs/clap/releases)
- [Changelog](https://github.com/clap-rs/clap/blob/master/CHANGELOG.md)
- [Commits](https://github.com/clap-rs/clap/compare/clap_complete-v3.0.4...clap_complete-v3.0.5)

---
updated-dependencies:
- dependency-name: clap_complete
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

chipbuster committed:
ci: Change branch detection method for crowdin (#3502)

AaronFriel committed:
feat: add username to Pulumi module (#3428)

* feat: add username to Pulumi module

* run dprint fmt

dependabot[bot] committed:
build: bump vuepress from 1.9.5 to 1.9.7 in /docs (#3494)

Bumps [vuepress](https://github.com/vuejs/vuepress/tree/HEAD/packages/vuepress) from 1.9.5 to 1.9.7.
- [Release notes](https://github.com/vuejs/vuepress/releases)
- [Changelog](https://github.com/vuejs/vuepress/blob/master/CHANGELOG.md)
- [Commits](https://github.com/vuejs/vuepress/commits/v1.9.7/packages/vuepress)

---
updated-dependencies:
- dependency-name: vuepress
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

matchai committed:
chore: resize and compress all flags

matchai committed:
docs: add snapcraft to installation instructions (#3513)

chipbuster committed:
chore: Update LICENSE file for 2022 (#3514)

dependabot[bot] committed:
build: bump serde from 1.0.135 to 1.0.136 (#3516)

Bumps [serde](https://github.com/serde-rs/serde) from 1.0.135 to 1.0.136.
- [Release notes](https://github.com/serde-rs/serde/releases)
- [Commits](https://github.com/serde-rs/serde/compare/v1.0.135...v1.0.136)

---
updated-dependencies:
- dependency-name: serde
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump clap from 3.0.12 to 3.0.13 (#3524)

Bumps [clap](https://github.com/clap-rs/clap) from 3.0.12 to 3.0.13.
- [Release notes](https://github.com/clap-rs/clap/releases)
- [Changelog](https://github.com/clap-rs/clap/blob/master/CHANGELOG.md)
- [Commits](https://github.com/clap-rs/clap/compare/v3.0.12...v3.0.13)

---
updated-dependencies:
- dependency-name: clap
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

KanchiShimono committed:
docs: remove meaningless invisible character (#3519)

dependabot[bot] committed:
build: bump toml_edit from 0.13.0 to 0.13.1 (#3523)

Bumps [toml_edit](https://github.com/ordian/toml_edit) from 0.13.0 to 0.13.1.
- [Release notes](https://github.com/ordian/toml_edit/releases)
- [Changelog](https://github.com/ordian/toml_edit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ordian/toml_edit/compare/v0.13.0...v0.13.1)

---
updated-dependencies:
- dependency-name: toml_edit
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump toml_edit from 0.13.1 to 0.13.2 (#3532)

Bumps [toml_edit](https://github.com/ordian/toml_edit) from 0.13.1 to 0.13.2.
- [Release notes](https://github.com/ordian/toml_edit/releases)
- [Changelog](https://github.com/ordian/toml_edit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ordian/toml_edit/compare/v0.13.1...v0.13.2)

---
updated-dependencies:
- dependency-name: toml_edit
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

CLIDragon committed:
feat: starship bug-report sets syntax highlighting for config file (#3529)

Set the syntax highlighting for the config file in starship bug-report, using the triple backtick syntax.

davidkna committed:
fix(elvish): use `$pwd` for `logical-path` (#3534)

The `$PWD` environment variable used by starship is not updated on startup by `elvish`. `elvish` also provides a separate `$pwd` variable that does get updated and seems to be be a better indicator for the current logical path.

Lyndeno committed:
feat(cmd_duration): Make notification timeout configurable (#3515)

* Allow customization of notification timeout

* Document new notification duration option

* Check for out-of-bounds timeout and correct it

* Implement ModuleConfig for u32

* Revert "Check for out-of-bounds timeout and correct it"

This reverts commit 52109ab5f7c336b55c81bccafb3adbfc81514553.

* Switch notification_timeout to u32

* Note notification_daemons might not honor timout

* Notification timeout defaults to daemon timeout

* Leave default value of notification_timeout blank in docs

TrickyPi committed:
fix(nodejs): use e718 as the default of symbol in node configuration (#3533)

* fix: use e718 as the default of symbol in node configuration

* wip: change nodejs symbol in docs/config & add a nodejs symbol configuration in docs/presets

* wip: update CONTRIBUTING.md

chipbuster committed:
fix(bash): Restore previous exit status in bash init (#3521)

* fix: Restore previous exit status in bash init

* Do it correctly this time

quickstar committed:
feat(localip): add module to print the current ipv4 address (#3289)

dependabot[bot] committed:
build: bump toml_edit from 0.13.2 to 0.13.3 (#3542)

Bumps [toml_edit](https://github.com/ordian/toml_edit) from 0.13.2 to 0.13.3.
- [Release notes](https://github.com/ordian/toml_edit/releases)
- [Changelog](https://github.com/ordian/toml_edit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ordian/toml_edit/compare/v0.13.2...v0.13.3)

---
updated-dependencies:
- dependency-name: toml_edit
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dodgez committed:
docs(preset): Fix redundant space in duration bracket preset (#3548)

* fix: redundant space in duration bracket preset

* revert: translated preset docs changes

This reverts commit be3349324ba5d4a70c7ec88bcc0f580706d0529b.

Co-authored-by: Zachary Dodge <zdodge@amazon.com>

schrieveslaach committed:
feat: Enable Undistract Me for Linux (#3547)

In #1019 the undistract feature has been added to starship without
enabling it by default due to the system dependency on libdbus. With
recent versions of notify-rust the dependency is no longer required and
the feature can be enabled by default.

dependabot[bot] committed:
build: bump toml_edit from 0.13.3 to 0.13.4 (#3549)

Bumps [toml_edit](https://github.com/ordian/toml_edit) from 0.13.3 to 0.13.4.
- [Release notes](https://github.com/ordian/toml_edit/releases)
- [Changelog](https://github.com/ordian/toml_edit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ordian/toml_edit/compare/v0.13.3...v0.13.4)

---
updated-dependencies:
- dependency-name: toml_edit
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump clap from 3.0.13 to 3.0.14 (#3552)

Bumps [clap](https://github.com/clap-rs/clap) from 3.0.13 to 3.0.14.
- [Release notes](https://github.com/clap-rs/clap/releases)
- [Changelog](https://github.com/clap-rs/clap/blob/master/CHANGELOG.md)
- [Commits](https://github.com/clap-rs/clap/compare/v3.0.13...v3.0.14)

---
updated-dependencies:
- dependency-name: clap
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

davidkna committed:
docs: remove stray `:::` (#3559)

chipbuster committed:
fix: Correctly detect older versions of powershell in bug-report (#3543)

* Factor out the shell version function

* Correct command to trim spaces + header

marlonrichert committed:
refactor: Improve compatibility with Zsh prompt theme system (#3480)

Zsh's `promptinit` expects a theme's hook functions to be named `prompt_<theme>_<hook>`. See https://github.com/zsh-users/zsh/blob/2876c25a28b8052d6683027998cc118fc9b50157/Functions/Prompts/promptinit#L155

segevfiner committed:
feat(pwsh): Set ExtraPromptLineCount (#3439)

* feat(pwsh): Set ExtraPromptLineCount

Fixes #3419

* Rename $prompt to $promptText

moko256 committed:
fix(status): Enable to convert from i64 to hex_status by casting instead of parsing status. (#3462)

* fix(status): Enable to convert from i64 to hex_status by casting instead of parsing status.

* Apply comment to src/context.rs

Co-authored-by: David Knaack <davidkna@users.noreply.github.com>

* Update README.md in configuration

Co-authored-by: David Knaack <davidkna@users.noreply.github.com>

CoelacanthusHex committed:
refactor(zsh): using add-zsh-hook instead of add hook manually (#3537)

[1]: https://github.com/starship/starship/pull/3480#discussion_r790280291
[2]: https://zsh.sourceforge.io/Doc/Release/User-Contributions.html#Writing-Themes

Signed-off-by: Coelacanthus <coelacanthus@outlook.com>

cg-jl committed:
feat(cli): Print arguments if argument parsing fails (#3560)

* fix(#3554): Print the command line argv on clap error

This is a very bare implementation that just prints the error
and then a note with the arguments passed, it does this manually
and doesn't use clap. I've also chosen to use `Vec`'s `Debug`
implementation instead of rolling my own one because I thought it was
good enough, but there might be a better way of doing all this.

Altogether, I think this will be very useful to help in the diagnostic
of other bugs :)

* fix(#3554): Print the command line argv on clap error

This is a very bare implementation that just prints the error
and then a note with the arguments passed, it does this manually
and doesn't use clap. I've also chosen to use `Vec`'s `Debug`
implementation instead of rolling my own one because I thought it was
good enough, but there might be a better way of doing all this.

Altogether, I think this will be very useful to help in the diagnostic
of other bugs :)

EDIT: removed `dbg!`, set it to exit always.

* correctness(exit): don't print argv / exit with error on help and
version error kinds

* fix: Avoid panicking when stdout/stderr closing unexpectedly

* refactor(cli): use `use_stderr` instead of manual match for error kinds

`clap` uses `use_stderr` to reliably check whether the error given is
actually an error coming from user input or rather a hint to display
other info (version, help, etc.)

Also reworded/moved a couple of comments so that they explain better
what is the thought process behind the code

dependabot[bot] committed:
build: bump semver from 1.0.4 to 1.0.5 (#3577)

Bumps [semver](https://github.com/dtolnay/semver) from 1.0.4 to 1.0.5.
- [Release notes](https://github.com/dtolnay/semver/releases)
- [Commits](https://github.com/dtolnay/semver/compare/1.0.4...1.0.5)

---
updated-dependencies:
- dependency-name: semver
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump notify-rust from 4.5.5 to 4.5.6 (#3575)

Bumps [notify-rust](https://github.com/hoodie/notify-rust) from 4.5.5 to 4.5.6.
- [Release notes](https://github.com/hoodie/notify-rust/releases)
- [Changelog](https://github.com/hoodie/notify-rust/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hoodie/notify-rust/compare/v4.5.5...v4.5.6)

---
updated-dependencies:
- dependency-name: notify-rust
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump shell-words from 1.0.0 to 1.1.0 (#3572)

Bumps [shell-words](https://github.com/tmiasko/shell-words) from 1.0.0 to 1.1.0.
- [Release notes](https://github.com/tmiasko/shell-words/releases)
- [Commits](https://github.com/tmiasko/shell-words/compare/v1.0.0...v1.1.0)

---
updated-dependencies:
- dependency-name: shell-words
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump os_info from 3.1.0 to 3.2.0 (#3576)

Bumps [os_info](https://github.com/stanislav-tkach/os_info) from 3.1.0 to 3.2.0.
- [Release notes](https://github.com/stanislav-tkach/os_info/releases)
- [Changelog](https://github.com/stanislav-tkach/os_info/blob/master/CHANGELOG.md)
- [Commits](https://github.com/stanislav-tkach/os_info/compare/v3.1.0...v3.2.0)

---
updated-dependencies:
- dependency-name: os_info
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump clap_complete from 3.0.5 to 3.0.6 (#3574)

Bumps [clap_complete](https://github.com/clap-rs/clap) from 3.0.5 to 3.0.6.
- [Release notes](https://github.com/clap-rs/clap/releases)
- [Changelog](https://github.com/clap-rs/clap/blob/master/CHANGELOG.md)
- [Commits](https://github.com/clap-rs/clap/compare/clap_complete-v3.0.5...clap_complete-v3.0.6)

---
updated-dependencies:
- dependency-name: clap_complete
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump unicode-segmentation from 1.8.0 to 1.9.0 (#3573)

Bumps [unicode-segmentation](https://github.com/unicode-rs/unicode-segmentation) from 1.8.0 to 1.9.0.
- [Release notes](https://github.com/unicode-rs/unicode-segmentation/releases)
- [Commits](https://github.com/unicode-rs/unicode-segmentation/commits)

---
updated-dependencies:
- dependency-name: unicode-segmentation
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

matchai committed:
docs(i18n): new Crowdin updates (#3460)

matchai committed:
chore(release): v1.3.0

abekoh committed:
fix: typo of variable in CONTRIBUTING (#3595)

Shan-Zhou committed:
fix(pwsh): Use global:error[0] for most recent error in powershell (#3584)

* Use global:error[0] for most recent error

The current version use $error[0], which is for the module instead of the global status and is always $null, so the $lastExitCodeForPrompot is always $origLastExitCode (see https://github.com/starship/starship/issues/3485 for a related bug).
It should be replaced with $global:error[0] 
Comments are also updated

* fix(pwsh): make Semantic PR bot happy

davidkna commented on src/modules/dolt_branch.rs:
```diff
@@ -0,0 +1,186 @@
+        return None;
+    }
+
+    if config.disabled {
+        return None;
+    }
+
```
This is unnecessary unless the module is disabled by default:
```suggestion
```

davidkna commented on src/configs/dolt_branch.rs:
```diff
@@ -0,0 +1,31 @@
+            style: "bold purple",
+            truncation_length: std::i64::MAX,
+            truncation_symbol: "…",
+            detect_file: Vec::default(),
+            detect_folder: vec![".dolt"],
```
Please add `detect_extensions`

davidkna commented on src/configs/dolt_branch.rs:
```diff
@@ -0,0 +1,31 @@
+    fn default() -> Self {
+        DoltBranchConfig {
+            format: "on [$symbol$branch]($style)",
+            symbol: " ",
```
There's some discussion around potentially switching to emojis only by default. Please preferably use an emoji symbol here, and please also update the presets per the [checklist](https://github.com/starship/starship/blob/master/CONTRIBUTING.md#new-module-checklist).

davidkna commented on src/modules/dolt_branch.rs:
```diff
@@ -0,0 +1,186 @@
+            })
+            .map(|variable| match variable {
+                "branch" => {
+                    let command = context.exec_cmd("dolt", &["branch", "--show-current"])?;
```
This commands seems to be very slow (~300ms). Is there a way to speed this up? This module also doesn't seem to work on Windows right now, but that could be due to #3582.

davidkna reviewed (changes requested):
Please also rebase to ensure the new Markdown formatting checks run.

dependabot[bot] committed:
build: bump follow-redirects from 1.14.6 to 1.14.8 in /docs (#3602)

Bumps [follow-redirects](https://github.com/follow-redirects/follow-redirects) from 1.14.6 to 1.14.8.
- [Release notes](https://github.com/follow-redirects/follow-redirects/releases)
- [Commits](https://github.com/follow-redirects/follow-redirects/compare/v1.14.6...v1.14.8)

---
updated-dependencies:
- dependency-name: follow-redirects
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

rashil2000 committed:
fix(fish): Emit clear-screen escape sequence only in left prompt (#3588)

davidkna committed:
fix: ignore empty `--jobs` argument (#3593)

abekoh committed:
Add option gcloud.project_aliases (#3599)

* feat: gcloud.project_aliases

* feat: add option gcloud.project_aliases

cubercsl committed:
fix(localip): disable localip module default (#3607)

dependabot[bot] committed:
build: bump serde_json from 1.0.78 to 1.0.79 (#3609)

Bumps [serde_json](https://github.com/serde-rs/json) from 1.0.78 to 1.0.79.
- [Release notes](https://github.com/serde-rs/json/releases)
- [Commits](https://github.com/serde-rs/json/compare/v1.0.78...v1.0.79)

---
updated-dependencies:
- dependency-name: serde_json
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

eltociear committed:
docs: fix typo in config/README.md (#3561)

seperator -> separator

pe committed:
docs: Rename Heading localip -> Local IP (#3612)

The lowercase l can easily be missread as a capital I, since all other headings are capitalised..

dependabot[bot] committed:
build: bump open from 2.0.2 to 2.0.3 (#3610)

Bumps [open](https://github.com/Byron/open-rs) from 2.0.2 to 2.0.3.
- [Release notes](https://github.com/Byron/open-rs/releases)
- [Changelog](https://github.com/Byron/open-rs/blob/main/changelog.md)
- [Commits](https://github.com/Byron/open-rs/compare/v2.0.2...v2.0.3)

---
updated-dependencies:
- dependency-name: open
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build: bump rand from 0.8.4 to 0.8.5 (#3615)

Bumps [rand](https://github.com/rust-random/rand) from 0.8.4 to 0.8.5.
- [Release notes](https://github.com/rust-random/rand/releases)
- [Changelog](https://github.com/rust-random/rand/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rust-random/rand/compare/0.8.4...0.8.5)

---
updated-dependencies:
- dependency-name: rand
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

pe committed:
docs: "Local IP" chapter in alphabetical order (#3613)

chipbuster committed:
fix: Exclude vuepress output from dprint (#3616)

alago1 committed:
fix(aws): Only display AWS if there are credentials configured (#3504)

* only display aws on credential_process defined

* add check for both credential_process and valid credentials

* fix tests

* update aws module documentation

* add better explanation of requirements to documentation

* add support for AWS_SECRET_ACCESS_KEY, AWS_SESSION_TOKEN, and update docs

* remove credential_process env var

v4nderstruck committed:
feat(kubernetes): add context user and cluster variables (#3569)

* added kubernetes context user, cluster + basic test

* updated docs

* docs format

* changed get_kube_ctx_component to return struct

davidkna committed:
build: bump clap from 3.0.14 to 3.1.0 (#3630)

* build: bump clap from 3.0.14 to 3.1.0

Bumps [clap](https://github.com/clap-rs/clap) from 3.0.14 to 3.1.0.
- [Release notes](https://github.com/clap-rs/clap/releases)
- [Changelog](https://github.com/clap-rs/clap/blob/master/CHANGELOG.md)
- [Commits](https://github.com/clap-rs/clap/compare/v3.0.14...clap_complete-v3.1.0)

---
updated-dependencies:
- dependency-name: clap
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* refactor(clap): fix deprecation warnings

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

segevfiner committed:
feat: Add a Windows application manifest (#3590)

* feat: Add a Windows application manifest

Closes #3589

* Enable longPathAware

* Switch to winres crate

* Only depend on winres on Windows

* Switch to cfg attribute

hpwxf committed:
docs(directory): add missing doc on option directory.repo_root_format (#3550)

* docs : add missing doc on option direcory.repo_root_format

* docs: fix doc format

dependabot[bot] committed:
build: bump clap_complete from 3.0.6 to 3.1.0 (#3636)

Bumps [clap_complete](https://github.com/clap-rs/clap) from 3.0.6 to 3.1.0.
- [Release notes](https://github.com/clap-rs/clap/releases)
- [Changelog](https://github.com/clap-rs/clap/blob/master/CHANGELOG.md)
- [Commits](https://github.com/clap-rs/clap/compare/clap_complete-v3.0.6...clap_complete-v3.1.0)

---
updated-dependencies:
- dependency-name: clap_complete
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
